IO_Function_checker.py

```diff
@@ -0,0 +1,43 @@
+"""checker for use of I/O functions.
+"""
+
+import astroid
+from pylint.interfaces import IAstroidChecker
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages
+
+
+class IOFunctionChecker(BaseChecker):
+
+    __implements__ = IAstroidChecker
+
+    name = 'I/O-functions'
+    msgs = {'E9998': ('Used I/O function %s',
+                      'I/O-function-not-allowed',
+                      'Used when you use the I/O functions "print", "open" or'
+                      '"input". Pylint just try to discourage this usage. '
+                      'That doesn\'t mean you can not use it !'),
+           }
+
+    @check_messages("I/O-function-not-allowed")
+    def visit_call(self, node):
+        if isinstance(node.func, astroid.Name):
+            name = node.func.name
+            # ignore the name if it's not a builtin (i.e. not defined in the
+            # locals nor globals scope)
+            if not (name in node.frame() or name in node.root()):
+                # if name in self.config.io_functions:
+                if name == "print" or name == "input" or name == "open":
```

Comment:
This check can be made a little cleaner by putting the forbidden names in a global variable. Try looking at what Hayley did if you aren't sure what I mean.